src/components/Categories.jsx

```diff
@@ -0,0 +1,56 @@
+import { Error } from "./Error";
+import { useState, useEffect } from "react";
+import axios from "axios";
+
+export const Categories = ({ nav }) => {
+  const [categories, setCategories] = useState([]);
+  const [categoryError, setCategoryError] = useState(false);
+
+  useEffect(() => {
+    (async function () {
+      try {
+        let categories = await axios.get("/api/categories");
+        setCategories(categories.data.categories);
+      } catch (err) {
+        setCategoryError(true);
+        console.log(err);
+      }
+    })();
+  }, []);
+  return (
+    <>
+      {nav === "nav" ? (
+        <>
+          {" "}
+          {categoryError && <Error err={"Categories can't be loaded"} />}
+          {categories.map(({ category, link }) => {
+            return (
+              <a
```

Comment:
I think you should use Link instead of <a> tag here.

Reply:
yeah i will edit that when i start working on features.